### ISSUE TITLE
ENH: add from_edges to IntervalIndex

### ISSUE THREAD

**Issue by Xiaoven** [pandas-dev/pandas#59641]
### Feature Type

- [X] Adding new functionality to pandas

- [ ] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

I wish I could use pandas to create an IntervalIndex like [0.0, 0.1), [0.1, 0.2), ..., [0.9, 1.0]. 

Currently, I successfully cut my data frame using the following code:
```py
bin_edges = [i / 10.0 for i in range(0, 11)]  # Create bins from 0.0 to 1.0 with a size of 0.1
df['bin'] = pd.cut(df['ratio'], bins=bin_edges, include_lowest=True)

print(df['bin'].unique())
# Output:
# [(-0.001, 0.1], (0.9, 1.0], (0.4, 0.5], (0.8, 0.9], (0.3, 0.4], ..., (0.5, 0.6], (0.2, 0.3], (0.1, 0.2], (0.6, 0.7], NaN]
# Length: 11
# Categories (10, interval[float64, right]): [(-0.001, 0.1] < (0.1, 0.2] < (0.2, 0.3] < (0.3, 0.4] ... (0.6, 0.7] < (0.7, 0.8] < (0.8, 0.9] < (0.9, 1.0]]
```

Then, I want to reindexing to add back the missing missing bins and fills them with zero counts using `fillna(0)` with the following code
```py
bin_counts = df['bin'].value_counts().reindex(pd.IntervalIndex.from_breaks(bin_edges)).fillna(0)
```
However, I found the `(-0.001, 0.1]` with a non-zero count is replaced by `(0, 0.1]` with a zero count.

I ask ChatGPT, and it recommends to use `bin_counts = df['bin'].value_counts().reindex(pd.IntervalIndex.from_edges(bin_edges)).fillna(0)` which lead to `AttributeError: type object 'IntervalIndex' has no attribute 'from_edges'`

### Feature Description

Add a new function to IntervalIndex, from_edges, to return IntervalIndex based on bin edges.
```
 def from_breaks(cls, bins) -> IntervalIndex
```

### Alternative Solutions

I haven't found one yet

**Comment by mroeschke** [comment]
Thanks for the suggestion but I believe you want `from_breaks` https://pandas.pydata.org/docs/reference/api/pandas.IntervalIndex.from_breaks.html#pandas.IntervalIndex.from_breaks so closing

**Comment by Xiaoven** [comment]
> Thanks for the suggestion but I believe you want `from_breaks` https://pandas.pydata.org/docs/reference/api/pandas.IntervalIndex.from_breaks.html#pandas.IntervalIndex.from_breaks so closing


`from_breaks` does not fully meet my requirements for creating intervals. My expected intervals are `[0, 0.1), [0.1, 0.2), ..., [0.8, 0.9), [0.9, 1.0]`, with both 0 and 1 included in the range. To achieve this with `pd.cut`, I use `pd.cut(df['ratio'], bins=bin_edges, include_lowest=True)` to treat the first interval specially as `(-0.001, 0.1]`, ensuring that 0 is included.

However, when I attempt to use `pd.IntervalIndex.from_breaks(bin_edges, closed='left')`, the last interval becomes `[0.9, 1.0)`, which excludes 1.0.

When I try `pd.IntervalIndex.from_breaks(bin_edges, closed='right')`, the first interval is `(0.0, 0.1]`, which excludes 0.0.

Finally, using `pd.IntervalIndex.from_breaks(bin_edges, closed='both')` includes both 0 and 1 in the intervals, but this approach leads to overlapping intervals, which is not what I want.

Given these issues, I'm looking for a solution that allows me to include both 0 and 1 in the intervals without overlap. Is there an alternative method or adjustment that I can apply to `from_breaks` to achieve this?
